app/src/androidTest/java/com/geekskool/dhobi/Providers/DataProviderTest.java

```diff
@@ -0,0 +1,70 @@
+package com.geekskool.dhobi.Providers;
+
+import android.content.ContentUris;
+import android.content.ContentValues;
+import android.net.Uri;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+import android.test.ProviderTestCase2;
+
+import com.geekskool.dhobi.Contracts.Contract;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Calendar;
+
+import static com.geekskool.dhobi.Contracts.Contract.CourseEntry.COLUMN_DURATION;
+import static com.geekskool.dhobi.Contracts.Contract.CourseEntry.COLUMN_END_DATE;
+import static com.geekskool.dhobi.Contracts.Contract.CourseEntry.COLUMN_LOCATION;
+import static com.geekskool.dhobi.Contracts.Contract.CourseEntry.COLUMN_NAME;
+import static com.geekskool.dhobi.Contracts.Contract.CourseEntry.COLUMN_START_DATE;
+
+/**
+ * Created by manisharana on 22/2/17.
+ */
+@RunWith(AndroidJUnit4.class)
+public class DataProviderTest extends ProviderTestCase2<DataProvider> {
+
+    private ContentValues values;
+
+    public DataProviderTest() {
+        super(DataProvider.class, Contract.CONTENT_AUTHORITY);
+    }
+
+    @Before
+    @Override
+    public void setUp() throws Exception {
+        setContext(InstrumentationRegistry.getTargetContext());
+        super.setUp();
+        values = new ContentValues();
+        String cName = "SampleName";
+        int days = 10;
+        String loc = "bangalore";
+        Calendar cal = Calendar.getInstance();
+
+        values.put(COLUMN_NAME, cName);
+        values.put(COLUMN_START_DATE, cal.getTimeInMillis());
+        values.put(COLUMN_END_DATE, cal.getTimeInMillis() + 1000);
+        values.put(COLUMN_DURATION, days);
+        values.put(COLUMN_LOCATION, loc);
+    }
+
+    @Test
```

Comment:
Though I'm not actually implying you didn't do this here, I wanted to make a note for everyone:

Take a look at "Behaviour-Driven Development" and the Given-When-Then format for structuring tests. If this makes sense, then try writing tests *backwards*: Start with the Assertion (the "then"), move up to the the action ("when"), and finally write the setup ("given").

Try to keep tests to a single assertion, if possible.